### PR TITLE
Test labels addition, update, and subsequent retrieval after archival [BA-6252]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

File: cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala [hunk 1]
```diff
@@ -130,9 +130,16 @@ class CromwellClient(val cromwellUrl: URL,
     simpleRequest[WorkflowOutputs](outputsEndpoint(workflowId, args))
   }
 
-  def labels(workflowId: WorkflowId, headers: List[HttpHeader] = defaultHeaders)
+  def labels(workflowId: WorkflowId,
+             newLabelsOpt: Option[List[Label]] = None,
+             headers: List[HttpHeader] = defaultHeaders)
             (implicit ec: ExecutionContext): FailureResponseOrT[WorkflowLabels] = {
-    simpleRequest[WorkflowLabels](labelsEndpoint(workflowId), headers=headers)
+    newLabelsOpt match {
```
Comment (cjllanwarne) [inline cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: Rather than having an optional parameter which branches the function into a totally unrelated code path, I think I would just add a new method called `updateLabels` with a non-optional `newLabels` parameter.

File: centaur/src/main/scala/centaur/test/Test.scala [hunk 2]
```diff
@@ -517,6 +517,40 @@ object Operations extends StrictLogging {
...
+            _ <- IO.sleep(2.seconds)
+            recurse <- eventuallyComparisonWithLatestLabelsSucceeds(expected, submittedWorkflow, workflow)
+          } yield recurse
+        })
+      } yield ()
+    }
+
+    override def run: IO[Unit] = {
+      val labelsToAdd = List(Label("newlyAddedLabel", "Twas brillig, and the slithy toves did gyre and gimble in the wabe"))
+      val labelsToUpdate = List(Label("newlyAddedLabel", "All mimsy were the borogoves, and the mome raths outgrabe"))
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply centaur/src/main/scala/centaur/test/Test.scala]: https://en.wiktionary.org/wiki/bikeshedding

File: cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala (hunk 1 above)
Comment (grsterin) [reply cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: I don't have a strong preference here, but semantically, the methods name `labels` suggests that it's universal. But I can extract addition in the different method.

File: centaur/src/main/scala/centaur/test/Test.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (grsterin) [reply centaur/src/main/scala/centaur/test/Test.scala]: > https://en.wiktionary.org/wiki/bikeshedding

Should I be offended?

File: cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala (hunk 1 above)
Comment (grsterin) [reply cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: Extracted

File: centaur/src/main/scala/centaur/test/Test.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply centaur/src/main/scala/centaur/test/Test.scala]: Haha, no, I was just pointing out that of all the details @mcovarr  could have commented on in the PR, he managed to find the one that was entirely superficial... 😄

File: cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala [hunk 3]
```diff
@@ -130,9 +130,18 @@ class CromwellClient(val cromwellUrl: URL,
     simpleRequest[WorkflowOutputs](outputsEndpoint(workflowId, args))
   }
 
-  def labels(workflowId: WorkflowId, headers: List[HttpHeader] = defaultHeaders)
+  def labels(workflowId: WorkflowId,
```
Comment (cjllanwarne) [inline cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala]: If you feel strongly, you could call this `getLabels`. I'm not too worried about that though.

File: centaur/src/main/scala/centaur/test/Test.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (grsterin) [reply centaur/src/main/scala/centaur/test/Test.scala]: 😄
